src/corebluetooth/central_delegate.rs

```diff
@@ -340,6 +354,51 @@ pub mod CentralDelegate {
             delegate,
             CentralDelegateEvent::DiscoveredPeripheral(held_peripheral),
         );
+
+        let puuid_nsstring = ns::uuid_uuidstring(cb::peer_identifier(peripheral));
+        let puuid = Uuid::from_str(&NSStringUtils::string_to_string(puuid_nsstring)).unwrap();
+
+        let manufacturer_data = ns::dictionary_objectforkey(_adv_data, unsafe {
+            cb::ADVERTISEMENT_DATA_MANUFACTURER_DATA_KEY
+        });
+        if manufacturer_data != nil {
+            // manufacturer_data: NSData
+            let length = ns::data_length(manufacturer_data);
+            if length >= 2 {
+                let bytes = ns::data_bytes(manufacturer_data);
+                let v = unsafe { slice::from_raw_parts(bytes, length as usize).to_vec() };
```

Comment:
Seeing as all you are doing with v is splitting it into two slices which you then copy, there's no need for the `to_vec()` here.